### PR TITLE
[WEB-4082] 15.0.2 - General styling updates, add Radio buttons

### PR DESCRIPTION
**Pull request by jamiehenson**
## Jira Ticket Link / Motivation

More supporting work for the docs ably-ui update.

## Summary of changes

Fixes all of the things in the dev column in the design team Asana [here](https://app.asana.com/0/1202054865216493/1205115927126178) (screenshotted to capture state)

Screenshot 2024-11-29 at 11 21 06

Also adds radio button classes to ably-ui as we didn't have them before - with this comes a reorganisation of the radio button and checkbox sections of the Forms Storybook section. Dark mode support has been added to a few things, and I've removed the separate `gui.css` file in favour of better reusing the classes we already had.

## How do you manually test this?

https://ably.github.io/ably-ui/

## Merge/Deploy Checklist

- [ ] Written automated tests for implemented features/fixed bugs
- [ ] Rebased and squashed commits
- [ ] Commits have clear descriptions of their changes
- [ ] Checked for any performance regressions

## Frontend Checklist

- [ ] No frontend changes in this PR
- [ ] Added before/after screenshots for changes
- [ ] Tested on different platforms/browsers with [Browserstack](https://www.browserstack.com)
- [ ] Compared with the initial design / our [brand guidelines](https://www.figma.com/file/jTQgyIovrhGBOf4Zrvjvbk/Ably-Linear-Design?node-id=118%3A0)
- [ ] Checked the code for accessibility issues ([VoiceOver User Guide](https://support.apple.com/en-gb/guide/voiceover/welcome/mac))?

## Release Notes

- **New Features**
  - Introduced a new radio button component with default and disabled states.
  - Added dark mode styles for various components, enhancing visual consistency.

- **Improvements**
  - Simplified input and dropdown components by removing dark mode variants.
  - Updated button styles, including hover and active states for better visual feedback.

- **Styling Changes**
  - Renamed CSS classes for better clarity and consistency.
  - Enhanced form component styles for improved user experience across light and dark themes.

- **Configuration Updates**
  - Adjusted Tailwind CSS configurations for font sizes and padding to reflect new design standards.

### CODE REVIEW

File: tailwind.config.js [hunk 1]
```diff
@@ -36,11 +36,12 @@ module.exports = {
...
       btn1: ["var(--fs-btn1)", "var(--lh-normal)"],
       btn2: ["var(--fs-btn2)", "var(--lh-tight)"],
-      btn3: ["var(--fs-btn2)", "var(--lh-tight)"],
-      btn4: ["var(--fs-btn3)", "var(--lh-dense)"],
+      btn3: ["var(--fs-btn3)", "var(--lh-tight)"],
+      btn4: ["var(--fs-btn4)", "var(--lh-dense)"],
       menu1: ["var(--fs-menu1)"],
       menu2: ["var(--fs-menu2)"],
       menu3: ["var(--fs-menu3)"],
+      menu4: ["var(--fs-menu4)"],
```
[the remark this answers is not in the document]
Comment (jamiehenson) [reply tailwind.config.js:44]: Don't worry mate it's used elsewhere

File: src/core/styles/Forms.stories.tsx [hunk 2]
```diff
@@ -194,6 +153,118 @@ export const Checkboxes = {
...
+            className="text-neutral-1300 dark:text-neutral-000"
+            htmlFor="radio-example-6"
+          >
+            Finest option
+          </label>
+        </div>
+      </div>
+    </div>
+  ),
+};
```
[the remark this answers is not in the document]
Comment (jamiehenson) [reply src/core/styles/Forms.stories.tsx]: Done that

**Review by kennethkalmer — CHANGES_REQUESTED** [review]
Something is amiss with the checkboxes in dark mode ;)

Forms - Docs ⋅ Storybook 2024-11-29 15-59-12

File: src/core/styles/legacy-buttons.css [hunk 3]
```diff
@@ -1,8 +1,8 @@
 @layer components {
   .ui-btn {
     @apply text-white bg-cool-black text-btn2 font-sans font-bold inline-block rounded p-btn;
-    @apply hover:text-white hover:bg-active-orange;
-    @apply active:text-white active:bg-red-orange;
+    @apply hover:text-white hover:bg-neutral-1100;
```
[the remark this answers is not in the document]
Comment (kennethkalmer) [reply src/core/styles/legacy-buttons.css:4]: Button (legacy) - Docs ⋅ Storybook 2024-11-29 16-00-23

[the remark this answers is not in the document]
Comment (jamiehenson) [reply src/core/styles/legacy-buttons.css:4]: Chef Watson's request

[the remark this answers is not in the document]
Comment (jamiehenson) [reply src/core/styles/legacy-buttons.css:4]: Worth bearing in mind that the legacy buttons aren't expected to have any dynamic dark mode support

**Comment by jamiehenson** [comment]
@kennethkalmer updated the forms Storybook page, good to be consistent with new additions. Alas not on the Github pages instance as I have the Toggle/TabMenu components on there and those are a greater review priority for now

**Review by kennethkalmer — APPROVED** [review]
Nice one @jamiehenson 👏

**Pull request merged**